src/transformStateWithClones.js

```diff
@@ -8,6 +8,51 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const statesHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = addProperties(currentState, action.extraData);
+        statesHistory.push(currentState);
+        break;
+      case 'removeProperties':
+        currentState = removeProperties(currentState, action.keysToRemove);
+        statesHistory.push(currentState);
+        break;
+      case 'clear':
+        currentState = clearProperties(currentState);
```

Comment:
[CODE STYLE] - Instead of reassigning `currentState` inside each case, you can directly modify the state copy without the need for the helper functions. This will make your code cleaner and avoid unnecessary function calls.